src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -322,30 +325,87 @@ uint32_t ConnectionHandlerImpl::OnSessionStartedCallback(
 #endif  // ENABLE_SECURITY
                         << " service " << static_cast<int>(service_type)
                         << " for session " << static_cast<int>(session_id));
-      return 0;
+      NotifySessionStartedFailure(is_protected);
+      return;
     }
     new_session_id = session_id;
-    if (hash_id) {
-      *hash_id = protocol_handler::HASH_ID_NOT_SUPPORTED;
-    }
+    hash_id = protocol_handler::HASH_ID_NOT_SUPPORTED;
   }
   sync_primitives::AutoReadLock read_lock(connection_handler_observer_lock_);
   if (connection_handler_observer_) {
     const uint32_t session_key = KeyFromPair(connection_handle, new_session_id);
-    const bool success = connection_handler_observer_->OnServiceStartedCallback(
-        connection->connection_device_handle(), session_key, service_type);
-    if (!success) {
-      LOG4CXX_WARN(logger_,
-                   "Service starting forbidden by connection_handler_observer");
-      if (protocol_handler::kRpc == service_type) {
-        connection->RemoveSession(new_session_id);
-      } else {
-        connection->RemoveService(session_id, service_type);
-      }
-      return 0;
+
+    ServiceStartedContext context(connection_handle,
+                                  session_id,
+                                  new_session_id,
+                                  service_type,
+                                  hash_id,
+                                  is_protected);
+    start_service_context_queue_.push(context);
+
+    connection_handler_observer_->OnServiceStartedCallback(
+        connection->connection_device_handle(),
+        session_key,
+        service_type,
+        params);
+  } else {
+    if (protocol_handler_) {
+      std::vector<std::string> empty;
+      protocol_handler_->NotifySessionStartedResult(
+          new_session_id, hash_id, is_protected, empty);
+    }
+  }
+}
+
+void ConnectionHandlerImpl::NotifyServiceStartedResult(
+    bool result, std::vector<std::string>& rejected_params) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ServiceStartedContext context;
+  bool available = start_service_context_queue_.pop(context);
```

Comment:
Yes thanks, fixed in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/8ccf8031fa494c021005d6a2a206d8ef81c985fa.